src/main/java/gyro/aws/lambda/FunctionResource.java

```diff
@@ -529,7 +530,7 @@ public void copyFrom(FunctionConfiguration configuration) {
         setTimeout(configuration.timeout());
         setMemorySize(configuration.memorySize());
         setTrackingConfig(configuration.tracingConfig() != null ? configuration.tracingConfig().modeAsString() : null);
-        setKmsKeyArn(configuration.kmsKeyArn());
+        setKmsKey(findById(KmsKeyResource.class, configuration.kmsKeyArn()));
```

Comment:
This requires a null check as kmsKey is not a required field
```suggestion
        setKmsKey(!ObjectUtils.isBlank(configuration.kmsKeyArn()) ? findById(KmsKeyResource.class, configuration.kmsKeyArn()) : null);
```